src/drivers/pca9685_pwm/pca9685_pwm.cpp

```diff
@@ -0,0 +1,598 @@
+}
+
+// TODO
+int PWMDriverWrapper::ioctl(cdev::file_t *filep, int cmd, unsigned long arg)
+{
+	int ret = OK;
+
+	lock();
+
+	switch (cmd) {
+	case MIXERIOCRESET:
+		_mixing_output.resetMixerThreadSafe();
+
+		break;
+
+	case MIXERIOCLOADBUF: {
+			const char *buf = (const char *)arg;
+			unsigned buflen = strlen(buf);
+
+			ret = _mixing_output.loadMixerThreadSafe(buf, buflen);
+
+			break;
+		}
+
+	default:
+		ret = -ENOTTY;
+		break;
+	}
+
+	unlock();
+
+	if (ret == -ENOTTY) {
+		ret = CDev::ioctl(filep, cmd, arg);
+	}
+
+	return ret;
+}
+
+int PWMDriverWrapper::print_usage(const char *reason)
+{
+	if (reason) {
+		PX4_WARN("%s\n", reason);
+	}
+
+	PRINT_MODULE_DESCRIPTION(
+		R"DESCR_STR(
+### Description
+This module is responsible for generate pwm pulse with PCA9685 chip.
+
+It listens on the actuator_controls topics, does the mixing and writes the PWM outputs.
+
+### Implementation
+This module depends on ModuleBase and OutputModuleInterface.
+IIC communication is based on CDev::I2C
+
+### Examples
+It is typically started with:
+$ linux_pwm start -a 64 -b 1
+
+Use the `mixer` command to load mixer files.
+`mixer load /dev/pca9685 ROMFS/px4fmu_common/mixers/quad_x.main.mix`
+)DESCR_STR");
+
+    PRINT_MODULE_USAGE_NAME("linux_pwm_output", "driver");
+    PRINT_MODULE_USAGE_COMMAND_DESCR("start", "Start the task");
+    PRINT_MODULE_USAGE_PARAM_INT('a',64,0,255,"device address on this bus",true);
+    PRINT_MODULE_USAGE_PARAM_INT('b',1,0,255,"bus that pca9685 is connected to",true);
+    PRINT_MODULE_USAGE_DEFAULT_COMMANDS();
+
+    return 0;
+}
+
+int PWMDriverWrapper::custom_command(int argc, char **argv) { // only for test use
+    static PCA9685 *pca= nullptr;
+    if (strcmp(argv[0], "teston") == 0){
+        if(pca==nullptr){
+            PX4_INFO("start pca9685 test");
+            pca=new PCA9685(1,0x40);
+            pca->Start();
+            pca->setFreq(50);
+            uint16_t out=1000;
+            pca->updatePWM(&out,1);
+        }
+    }
+
+    if (strcmp(argv[0], "testoff") == 0){
+        if(pca!=nullptr){
+            pca->Stop();
+            delete pca;
+            pca= nullptr;
+            PX4_INFO("stop pca9685 test");
+        }
+    }
+
+    if (strcmp(argv[0], "testfreq") == 0){
+        if(pca!=nullptr){
+            pca->setFreq(atoi(argv[1]));
+            PX4_INFO("change pca9685 freq");
+        }
+    }
+
+    if (strcmp(argv[0], "testpwm") == 0){
+        if(pca!=nullptr){
+            uint16_t pwm=atoi(argv[1]);
+            pca->updatePWM(&pwm,1);
+            PX4_INFO("change pca9685 freq");
+        }
+    }
+    return PX4_OK;
+}
+
+int PWMDriverWrapper::task_spawn(int argc, char **argv) {
+
+    auto *instance = new PWMDriverWrapper();
+
+    if (instance!=nullptr) {
+        _object.store(instance);
+        _task_id = task_id_is_work_queue;
+
+        int ch;
+        opterr = 0; // turn off parse error output
+        optind=1; // reset parse index
+        int _Addr=PCA9685_DEFAULT_ADDRESS;
```

Comment:
No leading underscore `_` for local variables. And variables should all be lower case.